src/libraries/System.Private.Xml/src/System/Xml/Schema/XmlSchema.cs

```diff
@@ -88,35 +89,41 @@ public class XmlSchema : XmlSchemaObject
             return parser.XmlSchema;
         }
 
+        [RequiresUnreferencedCode(XmlSerializer.LinkerSerializationMessage)]
         public void Write(Stream stream)
         {
             Write(stream, null);
         }
 
+        [RequiresUnreferencedCode(XmlSerializer.LinkerSerializationMessage)]
         public void Write(Stream stream, XmlNamespaceManager? namespaceManager)
         {
             XmlTextWriter xmlWriter = new XmlTextWriter(stream, null);
             xmlWriter.Formatting = Formatting.Indented;
             Write(xmlWriter, namespaceManager);
         }
 
+        [RequiresUnreferencedCode(XmlSerializer.LinkerSerializationMessage)]
         public void Write(TextWriter writer)
         {
             Write(writer, null);
         }
 
+        [RequiresUnreferencedCode(XmlSerializer.LinkerSerializationMessage)]
         public void Write(TextWriter writer, XmlNamespaceManager? namespaceManager)
         {
             XmlTextWriter xmlWriter = new XmlTextWriter(writer);
             xmlWriter.Formatting = Formatting.Indented;
             Write(xmlWriter, namespaceManager);
         }
 
+        [RequiresUnreferencedCode(XmlSerializer.LinkerSerializationMessage)]
         public void Write(XmlWriter writer)
         {
             Write(writer, null);
         }
 
+        [RequiresUnreferencedCode(XmlSerializer.LinkerSerializationMessage)]
```

Comment:
Can't we make this safe by preserving the correct things on `XmlSchema`?

Reply:
I think it should be possible - but it would be pretty tricky. There's probably a lot of dependencies which would have to be added between this method and all the various properties and other magic. The XmlSchema object model is quite large...

Reply:
Isn't it better for us to do it, than have a customer who needs this in a trimmed app be forced to try to figure it out themselves?

Reply:
I tried to stay away from intrusive code changes in this PR, just annotation. Happy to file a bug, though, if that's your preferred mechanism of tracking.

Reply:
> intrusive code changes

I wasn't thinking about intrusive code changes. I was thinking we would fix this with adding the necessary `[DynamicDependency]` attributes, plus a trimming test or two to ensure it works.

Reply:
I've addressed this by keeping all public members on XmlSchema and validated with a trimming test that I'm adding as well.